### ISSUE TITLE
When backdrop-filter is set on menus, text becomes blurry

### ISSUE THREAD

**Issue by NotWoods** [microsoft/fluentui#26764]
### Environment Information

- **Package version(s)**: @fluentui/react-components 9.13.0
- **Browser and OS versions**: Edge 112.0.1667.0, Windows 11

### Describe the issue:

We use acrylic menu designs in Loop, which requires the `backdrop-filter: blur()` CSS property. However, when mixed with `translate` this causes the menu text to become blurry. 

If inset properties (`left`, `top`) are used instead of a translate, then the blurriness disappears. Ideally we could make a menu that lets us change how the CSS is applied.

#### Actual behavior:

Text is blurred at 110% zoom

#### Expected behavior:

Text is clear.

### If applicable, please provide a codepen repro:
https://codesandbox.io/s/restless-voice-f50i0b?file=/example.tsx

### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? No
Requested priority: High

Products/sites affected: Microsoft Loop

**Comment by behowell** [comment]
Related to:
* #26038

That PR rounded position to whole pixel boundaries. However, it looks like applying other filters like `backdrop-filter: blur()` can result in the browser disabling pixel snapping. One potential fix is to round the position values to device pixels instead:
```ts
  const x = Math.round(coordinates.x * devicePixelRatio) / devicePixelRatio;
  const y = Math.round(coordinates.y * devicePixelRatio) / devicePixelRatio;
```

I'll make a PR with that change. I'm not entirely sure it's the right fix here, but it seems to fix the blurriness in the repro in this bug.

**Comment by ling1726** [comment]
It seems that Firefox had an issue like this and fixed on their end, I can't seem to repro it on that browser either https://bugzilla.mozilla.org/show_bug.cgi?id=1589198

The fix is fine, but we should probably log/find a bug on chromium and also test if this behaviour repros on Safari so we have the info for future reference

**Comment by NotWoods** [comment]
Filed https://bugs.chromium.org/p/chromium/issues/detail?id=1414915#c_ts1676310925

**Comment by behowell** [comment]
Re-closing. Already fixed by #26766, which takes devicePixelRatio into account when rounding the translate transform, so that the popover will land on device pixels to avoid blurriness.

**Comment by ling1726** [comment]
Reopening, because #26766 is being reverted by https://github.com/microsoft/fluentui/pull/27179. This is a browser bug especially since it only reproduces on chromium and not Firefox. The original fix to divide by device pixel ration causes blurriness for all cases that don't use backdrop-filter
